src/client/js/client.js

```diff
@@ -1138,17 +1139,85 @@ define([
+                for (updatePath in state.loading.changedNodes.update) {
+                    if (metaNodes.hasOwnProperty(updatePath)) {
+                        updatedMetaPaths.push(updatePath);
+                    }
+                }
+
+                if (updatedMetaPaths.length > 0) {
+                    // There are meta-nodes with updates.
+                    for (metaPath in metaNodes) {
+                        // For all meta nodes..
+                        if (metaNodes.hasOwnProperty(metaPath)) {
+                            for (i = 0; i < updatedMetaPaths.length; i += 1) {
+                                // check if it is a typeOf (includes mixins) any of the updated meta-nodes
+                                if (state.core.isTypeOf(metaNodes[metaPath],
+                                        metaNodes[updatedMetaPaths[i]]) === true) {
+                                    // if so add its path to the update nodes.
+                                    state.loading.changedNodes.update[metaPath] = true;
+                                }
+                            }
+                        }
+                    }
+                }
+                //console.log('Update after meta considered', Object.keys(state.loading.changedNodes.update));
+
+                // 2. Add Root node
+                state.loading.changedNodes.update[ROOT_PATH] = true;
+
+                // 3. Account for loads and unloads.
+                for (loadUnloadPath in state.loading.changedNodes.load) {
+                    if (state.loading.changedNodes.load.hasOwnProperty(loadUnloadPath)) {
+                        pathPieces = loadUnloadPath.split('/');
+                        while (pathPieces.length > 2) {
+                            pathPieces.pop();
+                            state.loading.changedNodes.update[pathPieces.join('/')] = true;
+                        }
+                    }
+                }
+
+                for (loadUnloadPath in state.loading.changedNodes.unload) {
+                    if (state.loading.changedNodes.unload.hasOwnProperty(loadUnloadPath)) {
+                        pathPieces = loadUnloadPath.split('/');
+                        while (pathPieces.length > 2) {
+                            pathPieces.pop();
+                            state.loading.changedNodes.update[pathPieces.join('/')] = true;
+                        }
+                    }
+                }
+
+                //console.log('Update after loads and unloads considered',
+                //    Object.keys(state.loading.changedNodes.update));
+            }
+
             for (i in state.nodes) {
                 if (state.nodes.hasOwnProperty(i)) {
-                    if (newerNodes[i]) {
-                        if (newerNodes[i].hash !== state.nodes[i].hash && state.nodes[i].hash !== '') {
+                    if (newerNodes[i] && newerNodes[i].hash !== state.nodes[i].hash && state.nodes[i].hash !== '') {

```

Comment:
I'm not sure, it's still being set to an integer. I haven't touched that part though.